### DISCUSSION TITLE
Удаление своства readOnly у компонента PrizmToggle

### DISCUSSION THREAD

**Discussion by ickisIckis** [zyfra/Prizm#2148] [ADR]
_Дата решения 01.11.2024._

### Мотив

На данный момент компонент PrizmToggle  имеет свойство readOnly. Оно запрещает пользователю изменять значение поля, а не взаимодействовать с ним и 
обычно предназначено для элементов с вводом данных (например, текстовых полей), но не для интерактивных компонентов, таких как Toggle - подробнее https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly#overview
В этом случае возникают сложности в понимании назначения readOnly, так как переключатели, по своей природе, ожидаются пользователями как интерактивные элементы. Это противоречит основной концепции "только для чтения", где ввод невозможен, но значение можно просмотреть и доводит понимание этого свойства до смешения с состоянием disabled.
В то же время разграничение состояний disabled и readOnly может привести к негативному пользовательскому опыту: пользователь видит, что элемент не отключен (не в состоянии disabled), но не может изменить его состояние, так как включено свойство readonly. Это ухудшает предсказуемость интерфейса.

### Краткое описание решения и его обоснование

Удалить свойство readOnly из компонента Toggle, оставив управление состоянием через свойство disabled. Это позволит избежать смешения с сотояниям disabled, упростить логику использования компонента, а также облегчит его api за счет удаления избыточного функионала.

#### Для этого:
1.  Перевести свойство в группу deprecated функционала в версиях Prizm 5+ и запланировать
2.  Запланировать его удаление в следующем мажорном релизе (6.0).

### Последствия

 + Компонент будет избавлен от свойства, которое для него не характерно.

 + Упрощение API компонента.

+  Улучшение пользовательского опыта.

 - Breaking change и возможный эффект на проекты.
 
 
#### Возможные альтернативы
Добавление специфической стилизации компонента для состояния readOnly

**Comment by PrizmDS** [comment]
Добро

**Comment by ickisIckis** [comment]
Связано с задачей #2047
Комментарий от дизайнера: https://github.com/zyfra/Prizm/issues/2047#issuecomment-2352054342

**Reply by ozknemoy** [reply]
поддерживаю идею
